### PR TITLE
Blob Heatsink

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Pretty much dies in one hit to bullets or melee, but is immune to emitter/laser fire. Costs 20 points.

Useful for planting in front of emitters/guarding your core from xrays.

Work in progress, just putting it up so people can pick apart the idea/balance before I bother finishing it.

### CODE REVIEW

**Comment by Robustin** [comment]
/shrug

Seems balanced but with juggernauts shitting on anyone who tries to melee right now I could see it just being another PITA for blobfighters without really adding anything to the mode.

**Comment by PKPenguin321** [comment]
:+1: 
blob rounds where the blob is 3 tiles from winning and then the core just vanishes because le x-ray from space are super lame

**Comment by Boredone** [comment]
So uh, does it actually STOP the X-Rays? Doesn't look like it code wise from what I can tell.

**Comment by PKPenguin321** [comment]
@Boredone 
> Work In Progress

**Comment by Boredone** [comment]
Ah, sorry, did not notice that part.

**Comment by Steelpoint** [comment]
This seems pretty neat, and is expensive enough to keep it as a more mid to late game thing.

File: code/game/gamemodes/blob/blobs/heatsink.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	atmosblock = 1
+
+/obj/effect/blob/heatsink/update_icon()
+	overlays.Cut()
+	color = null
+	var/image/I = new('icons/mob/blob.dmi', "blob")
+	if(overmind)
+		I.color = overmind.blob_reagent_datum.color
+	src.overlays += I
+	var/image/C = new('icons/mob/blob.dmi', "blob_heatskin_overlay")
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/heatsink.dm]: The icon in the file is named blob_heatink_overlay.
I'd suggest making the color of this overlay the `overmind.blob_reagent_datum.complementary_color` and maybe greyscaling it, to make it super visually distinct from nodes

**Comment by GunHog** [comment]
Name it "heatskin" or something. "heat sink" does not sound like an appropriate name for a biological entity.

File: code/game/gamemodes/blob/blobs/heatsink.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/obj/effect/blob/heatsink
+	name = "heatsink blob"
+	icon_state = "blank_blob"
+	desc = "A fleshy mass capable of asborbing laser fire."
+	health = 5
+	maxhealth = 5
+	atmosblock = 1
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/heatsink.dm]: You should override the fire_act() so it doesn't burn if you put it on top of a fire.

File: code/game/gamemodes/blob/blobs/heatsink.dm [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/obj/effect/blob/heatsink
+	name = "heatsink blob"
+	icon_state = "blank_blob"
+	desc = "A fleshy mass capable of asborbing laser fire."
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/blobs/heatsink.dm]: >asborbing

I'd suggest `A pulsating mass supported by thin tendrils, capable of absorbing laser fire.`

Comment (GunHog) [inline code/game/gamemodes/blob/blobs/heatsink.dm]: How about "a brightly glowing red mass", to be consistent with the other blob pieces? If the sprite had long, fin-like tendrils (similar to a real heat sink), you could describe it that way as well.

Comment (GunHog) [inline code/game/gamemodes/blob/blobs/heatsink.dm]: Wow, I did NOT see your comment there.

**Comment by ChangelingRain** [comment]
I'm not really against this, I might think it needs to cost a little more, but eh.
It should definitely have a HUD button, though.

**Comment by AdamElTablawy** [comment]
>blobber spam 
>heatsink

gggggg, even Xrays, the default counter to blob, cannot win anymore

**Comment by Cebasiz** [comment]
is it really wise to make the core de facto invincible by the time the blob is halfway towards an unstoppable superhealing player-controlled blobbernaut army

i mean Xrays don't even happen half the time and getting them should be a potent counter.

i don't think buffing blob even more is a wise idea, considering it already seems to win constantly.

**Comment by GunHog** [comment]
Do the "heat sinks" really stop the xray lasers? I personally do not think they should. They should resist damage from the laser itself, but should not stop it penetrating the rest of the blob.

**Comment by PKPenguin321** [comment]
i think they should, because a blob being super close to winning and getting supercucked by one science powergamer who stole a space suit is dumb
xray lasers will still work to do **lots** of damage, just not instakill the blob

**Comment by KorPhaeron** [comment]
Nevermind I don't want to finish this

**Comment by PKPenguin321** [comment]
korrrrr

**Pull request closed without merging**